From 5d3c73b58e404d392f8b9251a6ab084b26debdad Mon Sep 17 00:00:00 2001
From: "apronin@chromium.org" <apronin@chromium.org>
Date: Thu, 14 Jul 2016 17:29:40 -0700
Subject: [PATCH] tpm_tis_core: convert max timeouts from msec to jiffies

References: bsc#1020645, fate#321435, fate#321507, fate#321600, bsc#1034048, git-fixes aec04cbdf723
Patch-mainline: v4.8-rc1
Git-commit: 79b591c091b31291fecd9551db0a67fdc5901d50

tpm_tis_core was missing conversion from msec when assigning max
timeouts from constants.

Fixes: aec04cbdf723 ("tpm: TPM 2.0 FIFO Interface")
Signed-off-by: Andrey Pronin <apronin@chromium.org>
Reviewed-by: Jason Gunthorpe <jgunthorpe@obsidianresearch.com>
Reviewed-by: Jarkko Sakkinen <jarkko.sakkinen@linux.intel.com>
Signed-off-by: Jarkko Sakkinen <jarkko.sakkinen@linux.intel.com>
Acked-by: Michal Suchanek <msuchanek@suse.de>
---
 drivers/char/tpm/tpm_tis.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/char/tpm/tpm_tis.c b/drivers/char/tpm/tpm_tis.c
index 03a06b3..ace95b3 100644
--- a/drivers/char/tpm/tpm_tis.c
+++ b/drivers/char/tpm/tpm_tis.c
@@ -671,10 +671,10 @@ int tpm_tis_core_init(struct device *dev, struct tpm_tis_data *priv, int irq,
 		return -EIO;
 
 	/* Maximum timeouts */
-	chip->vendor.timeout_a = TIS_TIMEOUT_A_MAX;
-	chip->vendor.timeout_b = TIS_TIMEOUT_B_MAX;
-	chip->vendor.timeout_c = TIS_TIMEOUT_C_MAX;
-	chip->vendor.timeout_d = TIS_TIMEOUT_D_MAX;
+	chip->vendor.timeout_a = msecs_to_jiffies(TIS_TIMEOUT_A_MAX);
+	chip->vendor.timeout_b = msecs_to_jiffies(TIS_TIMEOUT_B_MAX);
+	chip->vendor.timeout_c = msecs_to_jiffies(TIS_TIMEOUT_C_MAX);
+	chip->vendor.timeout_d = msecs_to_jiffies(TIS_TIMEOUT_D_MAX);
 
 	if (wait_startup(chip, 0) != 0) {
 		rc = -ENODEV;
-- 
2.10.2

